src/quote/flashmint/nav/provider.ts

```diff
@@ -0,0 +1,125 @@
+import { BigNumber } from '@ethersproject/bignumber'
+import { Address } from 'viem'
+
+import { AddressZero } from 'constants/addresses'
+import { USDC } from 'constants/tokens'
+import { SwapQuoteProvider } from 'quote/swap'
+import {
+  Exchange,
+  getFlashMintNavContract,
+  isSameAddress,
+  slippageAdjustedTokenAmount,
+  SwapDataV3,
+} from 'utils'
+import { getRpcProvider } from 'utils/rpc-provider'
+
+import { QuoteProvider, QuoteToken } from '../../interfaces'
+import { getReserveAssetInputAmount } from './utils'
+
+export interface FlashMintNavQuoteRequest {
+  chainId: number
+  isMinting: boolean
+  inputToken: QuoteToken
+  outputToken: QuoteToken
+  // In contrast to other quote providers, we always need the input amount (not the index amount)
+  inputTokenAmount: BigNumber
+  slippage: number
+}
+
+export interface FlashMintNavQuoteQuote {
+  inputTokenAmount: BigNumber
+  outputTokenAmount: BigNumber
+  reserveAssetSwapData: SwapDataV3
+}
+
+export class FlashMintNavQuoteProvider
+  implements QuoteProvider<FlashMintNavQuoteRequest, FlashMintNavQuoteQuote>
+{
+  constructor(
+    private readonly rpcUrl: string,
+    private readonly swapQuoteProvider: SwapQuoteProvider
+  ) {}
+
+  async getQuote(
+    request: FlashMintNavQuoteRequest
+  ): Promise<FlashMintNavQuoteQuote | null> {
+    const {
+      chainId,
+      inputToken,
+      inputTokenAmount,
+      isMinting,
+      outputToken,
+      slippage,
+    } = request
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const usdc = USDC.address!
+
+    const swapQuoteRequest = {
+      chainId,
+      inputToken: isMinting ? inputToken.address : usdc,
+      outputToken: isMinting ? usdc : outputToken.address,
+      inputAmount: inputTokenAmount.toString(),
+      // TODO:
+      sources: [Exchange.UniV3],
```

Comment:
@edkim @pblivin0x should we use more than UniV3 here? First impressions from tests was that UniV3 were most reliable quotes tho.